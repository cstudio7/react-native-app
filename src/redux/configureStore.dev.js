import { createStore, applyMiddleware, compose } from 'redux';
import { createLogger } from 'redux-logger';
import rootReducer from './rootReducer/rootReducer';

const configureStore = preloadedState => {
  const store = createStore(
    rootReducer,
    preloadedState,
    compose(applyMiddleware(createLogger()))
  );

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./rootReducer/rootReducer', () => {
      const nextRootReducer = require('./rootReducer/rootReducer')
        .default;
      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
};

export default configureStore;
