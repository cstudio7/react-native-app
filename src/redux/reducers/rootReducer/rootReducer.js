import { combineReducers } from 'redux';
import names from '../names/names';

const rootReducer = combineReducers({
  names
});

export default rootReducer;
