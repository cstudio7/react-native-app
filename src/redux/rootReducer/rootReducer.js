import { combineReducers } from 'redux';
import names from '../namesReducer/reducer';

const rootReducer = combineReducers({
  names
});

export default rootReducer;
