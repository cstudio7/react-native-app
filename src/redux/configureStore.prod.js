import { createStore, applyMiddleware, compose } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage'; // defaults to localStorage for web and AsyncStorage for react-native
import amplitude from '../middleware/amplitude/amplitude';
import rootReducer from './rootReducer/rootReducer';

const configureStore = () => {
  const persistConfig = {
    key: 'root',
    storage
  };

  const persistedReducer = persistReducer(persistConfig, rootReducer);
  const store = createStore(
    persistedReducer,
    compose(applyMiddleware(amplitude))
  );
  const persistor = persistStore(store);

  return { store, persistor };
};

export default configureStore;
