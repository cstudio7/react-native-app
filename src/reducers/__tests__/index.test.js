import { createStore } from 'redux';

import rootReducer from '../';

const store = createStore(rootReducer);

it('checks that initial state of the root reducer matches what child reducers return given an empty action', () => {
  expect(store.getState()).toMatchSnapshot();
});