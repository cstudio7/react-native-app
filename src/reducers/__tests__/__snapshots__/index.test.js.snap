// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks that initial state of the root reducer matches what child reducers return given an empty action 1`] = `
Object {
  "names": Object {
    "female": Array [],
    "male": Array [],
  },
  "rate": Object {
    "rated": false,
    "visitsCount": 0,
  },
}
`;
